core/src/blocktree.rs

```diff
@@ -2939,9 +2959,15 @@ pub mod tests {
                     .cloned()
                     .map(|blob| Arc::new(RwLock::new(blob)))
                     .collect();
-                let mut coding_generator = CodingGenerator::new(Arc::clone(&blocktree.session));
+                let mut coding_generator = CodingGenerator::new(Arc::new(
+                    Session::new(
+                        blocktree.erasure_config.num_data(),
```

Comment:
erasure_config comes from blocktree for the generator?

Reply:
Yes, the validator config is passed up to blocktree.

Reply:
I don't get it...  each erasure block has its own config, dictated by the leader for the slot.  generator comes from node config, blocktree comes from the data plane (i.e. the blobs received from other leaders)

Reply:
Blocktree is initialized here with the node specific erasure configuration: https://github.com/solana-labs/solana/pull/5018/files#diff-a8d11c36b97d52f7aa49ba42e46c8c75R350

Reply:
ok, that's the wrong place

blocktree doesn't need to know the node's erasure config, only broadcast does